Yoast/yoast-components
# 9771/slug from prop instead of from state

## Summary

This PR can be summarized in the following changelog entry:

* Changed slug to prop slug.

## Relevant technical choices:

* Removed controlled input. Changed it to styled component.

## Test instructions

This PR can be tested by following these steps:

* see sibling pr. see parent

Fixes # https://github.com/Yoast/wordpress-seo/issues/9771
